### DISCUSSION TITLE
Plugin doesn't properly starts with greetd

### DISCUSSION THREAD

**Discussion by noahfraiture** [hyprwm/Hyprland#8356] [Q&A] [answered]
Hello,

I'm on nixos and I'm making some test, and it appears that if I use greetd with autologin, hyprspace failed to load. With this :
```
      greetd = lib.mkIf config.greetd.enable {
        enable = true;
        settings = rec {
          initial_session = {
            command = "${pkgs.hyprland}/bin/Hyprland";
            user = "noah";
          };
          default_session = initial_session;
        };
      };
```
It doens't work. But with 
```
      displayManager = {
        sddm = lib.mkIf config.sddm.enable {
          enable = true;
          theme = "${import ./sddm-theme.nix { inherit pkgs; }}";
        };
      };
```
It works. This is the only difference between my commit that works and the one that doesn't.
This is how I load my plugin in both situation 
```
    wayland.windowManager.hyprland = {
      plugins = [ inputs.hyprspace.packages.${pkgs.system}.Hyprspace ];
```

Is it a bug or an I missing something ?

**Comment by fufexan** [comment, marked as answer]
In greetd, you're setting the command to the nixpkgs Hyprland package, but you're using the flake package. As such, the plugin will not load on the nixpkgs version. If you only configure hyprland through HM, then you can safely pass `initial_command = "Hyprland";`.

**Reply by noahfraiture** [reply]
Oh right, thank you !
